python/tritonserver/_api/_allocators.py

```diff
@@ -238,24 +247,38 @@ def allocate(
         memory_type_id,
         _user_object,
     ):
-        if self._memory_type is not None:
-            memory_type = self._memory_type
-            memory_type_id = self._memory_type_id
-
-        memory_allocator = self._memory_allocator
-        if memory_allocator is None:
-            memory_allocator = default_memory_allocators[memory_type]
-
-        memory_buffer = memory_allocator.allocate(
-            byte_size, memory_type, memory_type_id
-        )
+        try:
+            if self._memory_type is not None:
+                memory_type = self._memory_type
+                memory_type_id = self._memory_type_id
+
+            memory_allocator = self._memory_allocator
+            if memory_allocator is None:
+                if memory_type in default_memory_allocators:
+                    memory_allocator = default_memory_allocators[memory_type]
+                else:
+                    LogMessage(
+                        LogLevel.WARN,
+                        f"Requested memory type {memory_type} is not supported, falling back to {MemoryType.CPU}",
+                    )
+                    memory_type = MemoryType.CPU
+                    memory_type_id = 0
+                    memory_allocator = default_memory_allocators[memory_type]
+
+            memory_buffer = memory_allocator.allocate(
+                byte_size, memory_type, memory_type_id
+            )
 
-        return (
-            memory_buffer.data_ptr,
-            memory_buffer,
-            memory_buffer.memory_type,
-            memory_buffer.memory_type_id,
-        )
+            return (
+                memory_buffer.data_ptr,
+                memory_buffer,
+                memory_buffer.memory_type,
+                memory_buffer.memory_type_id,
+            )
+        except Exception as e:
+            message = f"Catastrophic failure in allocator: {e}, returning NULL"
+            LogMessage(LogLevel.ERROR, message)
+            return (0, None, MemoryType.CPU, 0)
```

Comment:
Not sure if it's possible, but might be nice to have some namedtuple or something like `INVALID_MEMORY` for use/comparison purposes that is just this tuple under the hood (in the future)

Reply:
we could subclass from a tuple - or actually we can just create a singleton for invalid memory